SeedSigner/seedsigner
# [Enhancement] Preserve scroll position in long `ButtonListScreen`

## The Problem
Scroll through the list of Advanced settings. Select a setting to change (e.g. "Show xpub details"). Update that setting or use the BACK arrow. Upon returning to the Advanced settings list, the screen will have scrolled its list so that the setting you had just reviewed/changed will be rendered at the top. This behavior gets less and less desirable as you reach the end of the Advanced settings list ("Show partner logos").

Also happens in the Address Explorer's list of addrs.

## This PR
Alters the `SettingsMenuView` (with some minor support in `ButtonListScreen` and `SettingsEntryUpdateSelectionView`) to preserve its list state and re-render it with the same scroll position you left it at when you return.

Matching changes in `ToolsAddressExplorerAddressListView` (with support in `ToolsAddressExplorerAddressView`).

## How to test
* Test menu items that are already visible without any list scrolling (e.g. "Sig types"). Test the first item in the list, the last item on the page that is visible without scrolling, test middle items.
* Force the list to scroll and again test top item, bottom item, middle.
* Scroll to the bottom of the list and again test top, bottom, middle.
* Test selections that auto-return you to the list (enable/disable types), BACK arrow return to the list, multiselects that stay in place and require you to use the BACK arrow.
* Same tests in Address Explorer.

In each case, when you return to the Advanced settings list, your scroll position should be restored.

---

Note: I believe that the Advanced settings page and the Address Explorer are the only scrolling `ButtonListView`s that attempts to restore their previous position. The `SeedOptionsView`, for example, can be a scrolling list if enough options are enabled,  but always resets at the top when you return (which I think is fine).

If there *are* any other screens that attempt to preserve the scroll position, let me know so I can update them to use this technique.
